10-angular/resources_and_factories/readme.md

```diff
@@ -1,345 +1,521 @@
-# Angular Services and Factories
+# Factories, Services and `ngResource`
 
 ## Learning Objectives
 
-- Explain what dependency injection is and what problem it solves
-- Explain the purpose of services in Angular
-- Create a custom service to access data from an api
-- Explain the purpose of the $routeProvider in Angular
-- Create restful client-side routes using $routeProvider
-- Explain the purpose of templates in Angular
-- Use the ng-view directive to load angular templates
-- Define multiple controllers in a single module
-- Use $routeParams and $location to access query parameters and update the URL
-- Difference between Service and Factory.
-- Create separate views and routes for each CRUD action
-- Compare/contrast the components of angular to OOJS / Backbone.js
+* Explain the purpose of and differentiate between Factories and Services in Angular.
+* Use `ngResource` to pull information from an API.
+* Define multiple controllers in a single module.
+* Use $stateParams to access query parameters and update the URL.
+* Create separate views and routes for each CRUD action.
 
-## Overview of Today's Lesson (role of Templates, Routers, and Services)
+## Framing (2.5 minutes / 0:02)
 
-We'll:
+## Set Up Grumblr API (2.5 minutes / 0:05)
 
-- use a service so that we can connect to an API to store our data.
-- use templates to organize reusable chunks of HTML.
-- use multiple controllers to give context to a view.
-- use a router to read and update the url to represent state.
+Let's start by cloning and running a Grumblr Rails API in the background. Our front-end Grumblr application will make AJAX calls to this API.
 
-## Set up grumblr API
-
-    $ git clone git@github.com:ga-dc/grumblr_rails_api.git
-
-## Walkthrough of Current App
-
-    $ git clone git@github.com:ga-dc/grumblr_angular.git
+```bash
+$ git clone git@github.com:ga-dc/grumblr_rails_api.git
+$ cd grumblr_rails_api
+$ touch config/secrets.yml
+# Follow the instructions in the repo README re: secrets.yml
+$ bundle install
+$ rails s
+```
 
-Current Problems:
+## Walkthrough of Current App (20 minutes / 0:25)
 
-1. No url changes or representation of state.
-2. All HTML is in a single file.
-3. Data does not persist.
+We'll continue working on our front-end Grumblr Application from where you left off in your UI Router lesson. Follow either of the two instructions below to get your starter code.
```

Comment:
Not 100% accurate since we're asking them to do API-less CRUD for homework. Maybe:

> Where we're picking up the app, it has only a functioning `index` route that uses grumbles hardcoded into the index controller. Also, it doesn't use `$locationProvider` to remove the hashmark from URLs: all the URLs will be something like `index.html#/grumbles/32`. This makes things a bit easier for development.